contracts/LimitsChecker.sol

```diff
@@ -0,0 +1,267 @@
+    ) {
+        for (uint256 i = 0; i < _setLimitParametersRoleHolders.length; i++) {
+            _setupRole(SET_LIMIT_PARAMETERS_ROLE, _setLimitParametersRoleHolders[i]);
+        }
+        for (uint256 i = 0; i < _updateLimitSpendingsRoleHolders.length; i++) {
+            _setupRole(UPDATE_LIMIT_SPENDINGS_ROLE, _updateLimitSpendingsRoleHolders[i]);
+        }
+        bokkyPooBahsDateTimeContract = _bokkyPooBahsDateTimeContract;
+    }
+
+    // -------------
+    // EXTERNAL METHODS
+    // -------------
+
+    /// @notice Checks if _payoutSum is less than may be spent in the period
+    /// @param _payoutSum Motion sum
+    /// @param _motionDuration Motion duration - minimal time required to pass before enacting of motion
+    /// @return True if _payoutSum is less than may be spent in the period
+    function isUnderSpendableBalance(uint256 _payoutSum, uint256 _motionDuration)
+        external
+        view
+        returns (bool)
+    {
+        if (block.timestamp + _motionDuration >= currentPeriodEnd) {
+            return _payoutSum <= limit;
+        } else {
+            return _payoutSum <= limit - spent;
+        }
+    }
+
+    /// @notice Checks if _payoutSum is less than may be spent,
+    /// @notice updates period if needed and increases the amount spent in the current period
+    function updateSpendableBalance(uint256 _payoutSum)
+        external
+        onlyRole(UPDATE_LIMIT_SPENDINGS_ROLE)
+    {
+        _checkAndUpdateLimitParameters();
+        _checkLimit(_payoutSum);
+        _increaseSpent(_payoutSum);
+
+        (
+            uint256 _alreadySpentAmount,
+            uint256 _spendableAmount,
+            uint256 _periodStartTimestamp,
+            uint256 _periodEndTimestamp
+        ) = this.getCurrentPeriodState();
+
+        emit FundsSpent(
+            _alreadySpentAmount,
+            _spendableAmount,
+            _periodStartTimestamp,
+            _periodEndTimestamp
+        );
+    }
+
+    /// @notice Returns balance that can be spent in the current period
+    /// @return Balance that can be spent in the current period
+    function currentSpendableBalance() external view returns (uint256) {
+        return limit - spent;
+    }
+
+    /// @notice Sets periodDurationMonth and limit
+    /// currentPeriodEnd will be calculated as a calendar date of the beginning of next month,
+    /// bi-months, quarter, half year, or year period.
+    /// @param _limit Limit to set
+    /// @param _periodDurationMonth  Length of period in months to set.
+    /// Duration of the period can be 1, 2, 3, 6 or 12 months.
+    function setLimitParameters(uint256 _limit, uint256 _periodDurationMonth)
+        external
+        onlyRole(SET_LIMIT_PARAMETERS_ROLE)
+    {
+        _checkPeriodDurationMonth(_periodDurationMonth);
+        periodDurationMonth = uint64(_periodDurationMonth);
+        currentPeriodEnd = uint128(_getPeriodEndFromTimestamp(block.timestamp));
+        limit = uint128(_limit);
```

Comment:
there should be enforcement of `_limit <= type(uint128).max` before this assignment

Reply:
done!